### ISSUE TITLE
Enable alerts for iMammalia - wild boar carcass

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/MammalNet-Europe#87]
Summarising requirements discussed by email
- [ ] setup the option for trigger alerts for iMammalia survey.  Requires trigger template?
http://warehouse1.indicia.org.uk/index.php/trigger/edit/22

Refine for wild boar carcass only records
- [ ] write a custom trigger report that picks out records of boars which have an attribute defining them as a carcass. 

Enable alerts for the 2016 countries list which records index against

**Comment by johnvanbreda** [comment]
Plan is to enable a trigger report as a quick option, then later customise the species alerts system to allow filtering on custom attribute data (possibly requiring pre-defined custom configurations to simplify the setup for users).

I've implemented the trigger report approach. The way I did this, in order to get it working as quickly as possible, was to set up a notification trigger report on the warehouse which runs every 15 minutes in the background. When it detects a wild boar that has observation type dead (other or roadkill), it uses a custom lookup table to find the people to notify and if there is a country filter. This table is in the database and is called imammalia.dead_boar_notifications, so it’s necessary to run a query to insert a record into this table if anyone else requires a notification. The query looks like the following:

````sql
insert into imammalia.dead_boar_notifications(user_id, location_ids)
values (1234, ARRAY[5678]::integer[])
```

In this query, 1234 should be replaced with a warehouse user ID and 5678 replaced with the location ID of a country or other indexed location.

This doesn’t preclude us setting up a different mechanism where people can self-register for boar carcass alerts, but it was the quickest way I could get something running.
